### PR TITLE
Added: Extensible general settings page and timezone configuration

### PR DESCRIPTION
**Pull request by joeferner**
- [x] joeferner
- [x] mwizeman sfeng88
- [ ] joeybrk372 rygim jharwig EvanOxfeld

This change adds a general settings page to the user account modal dialog.
This change also adds a front end setting to change the timezone for all
front end times.

image

Testing Instructions:
- Have entities with time properties. Open the user account page and change the timezone. Verify that the times have changed correctly to the new timezone.

CHANGELOG
Added: Extensible general settings page to the account modal
Added: Front end timezone configuration

### CODE REVIEW

File: docs/extension-points/front-end/userAccount/index.md [hunk 1]
```diff
@@ -35,3 +35,42 @@ Remember to add a i18n value in a MessageBundle.properties. This will be display
...
+User Account Setting Plugin
+===========================
+
+Plugin to add a general setting to the user account settings page
+
+To register a boolean setting:
+
+```
+require([
+  'configuration/plugins/registry',
```
Comment (jharwig) [inline docs/extension-points/front-end/userAccount/index.md]: all examples should use `public/v1/api`, and I wouldn't recommend they require a file to get an enum. Just have them pass a string and validate it on registration

File: web/war/src/main/webapp/js/workspaces/userAccount/bundled/settings/Settings.jsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+     *
+     * @param {string} identifier Unique identifier for this setting
+     * @param {string} group The group i18n message id
+     * @param {string} displayName The display name i18n message id
+     * @param {string} type One of SettingType
+     */
+    registry.documentExtensionPoint('org.visallo.user.account.page.setting',
+        'Add new settings to the general settings page',
+        function (e) {
+            return ('identifier' in e) && ('group' in e) && ('displayName' in e) && ('type' in e);
```
Comment (jharwig) [inline web/war/src/main/webapp/js/workspaces/userAccount/bundled/settings/Settings.jsx]: Validate the `type` here

File: web/war/src/main/webapp/js/workspaces/userAccount/bundled/settings/Settings.jsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+], function (React, registry, SettingType, BooleanSetting) {
+    'use strict';
+
+    /**
+     * Adds new settings to the general settings page
+     *
+     * @param {string} identifier Unique identifier for this setting
+     * @param {string} group The group i18n message id
+     * @param {string} displayName The display name i18n message id
+     * @param {string} type One of SettingType
```
Comment (jharwig) [inline web/war/src/main/webapp/js/workspaces/userAccount/bundled/settings/Settings.jsx]: Add the possible options here: `BOOLEAN`, `custom`, etc

Comment (jharwig) [inline web/war/src/main/webapp/js/workspaces/userAccount/bundled/settings/Settings.jsx]: could also add `@param {string=} componentPath Used when type=custom`

File: web/war/src/main/webapp/js/workspaces/userAccount/bundled/settings/Settings.jsx (hunk 2 above)
Comment (jharwig) [inline web/war/src/main/webapp/js/workspaces/userAccount/bundled/settings/Settings.jsx]: Also verify componentPath is provided when type=custom

**Comment by joeferner** [comment]
@jharwig I made the changes you requested, take another look please

**Comment by EvanOxfeld** [comment]
@joeferner Looks like there are JS test failures.

**Pull request merged**